### PR TITLE
terra-navigation-side-menu : Keyboard navigation Fixes

### PR DESCRIPTION
**Pull request by MadanKumarGovindaswamy**
### Summary

**What was changed:**

Added Keyboard navigation.
Remove Action Header component

**Why it was changed:**

Keyboard navigation was not functioning.
Action header component is not required.

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [x] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10355 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Review by saket2403 — APPROVED** [review]

File: packages/terra-navigation-side-menu/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_medium/navigation-side-menu-spec/hover_navigation_side_menu_display.png
Comment (sugan2416) [inline packages/terra-navigation-side-menu/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_medium/navigation-side-menu-spec/hover_navigation_side_menu_display.png:1]: There is a difference in the background colour below the last item. Is this indication of reduced height of side menu? Is this expected ?

File: packages/terra-navigation-side-menu/src/NavigationSideMenu.module.scss
Comment (sugan2416) [inline packages/terra-navigation-side-menu/src/NavigationSideMenu.module.scss:1]: Don't we need UX design +1 for these ?

Comment (MadanKumarGovindaswamy) [reply packages/terra-navigation-side-menu/src/NavigationSideMenu.module.scss:1]: No we just replaced action header component. and still we have used same icon and css styles used in action header.

**Review by sugan2416 — APPROVED** [review]

**Comment by SwethaM03** [comment]
+1 for keyboard navigation of terra navigation side menu

**Pull request merged**